Core/Src/main.c

```diff
@@ -87,17 +89,46 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
+
+  uint8_t RxData[3];
+  uint8_t TxData[3] = {0x01, 0x80, 0x00};
+  uint16_t ADC_Value;
+  float PWM_DutyCycle; //
+  uint8_t PWM_DutyCycle_Count;
+  const uint16_t PERIOD = 0xfa00; // 64000, our period, in hexadecimal
+
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_6, GPIO_PIN_RESET); // Pulling CS line high
+	  HAL_SPI_TransmitReceive( &hspi1, RxData, TxData, sizeof(TxData), 500);
+	  // Above is ADC communication, 500ms was just a random number I thought was reasonable
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_6, GPIO_PIN_SET); // Pulling CS line low
+
+	  // Below is combining the necessary 10 bits of data we need into 1.
+	  ADC_Value = RxData[1] << 8;
+	  ADC_Value |= RxData[2];
```

Comment:
The bit-wise manipulations here are correct. But according to the datasheet, unexpected bits can occupy the first 6 bits in the `RxData[1]', so I would recommend masking the `ADC_Value` with a 10-bit mask to make sure only 10-bits are used from the rx_data.